nixos/modules/services/monitoring/prometheus/exporters/nextcloud.nix

```diff
@@ -58,12 +59,13 @@ in
           --addr ${cfg.listenAddress}:${toString cfg.port} \
           --timeout ${cfg.timeout} \
           --server ${cfg.url} \
-          ${if cfg.tokenFile == "" then ''
+          ${if cfg.passwordFile != null then ''
             --username ${cfg.username} \
```

Comment:
While you're at it, might as well make `username` optional as it isn't required for token-based auth

Reply:
hmm, that'd make sense, but username currently has a default value of 'nextcloud-exporter'. That default doesn't make sense anymore if we make the field optional, but removing it might break existing setups.

Reply:
I guess you're right

Reply:
ah I did not notice your suggestion because I didn't get an email notification of the edit :/ - updated